### DISCUSSION TITLE
Expandable navbar (sidebar)

### DISCUSSION THREAD

**Discussion by atumas-bananamilk** [mantinedev/mantine#4601] [Feature requests]
Could we get an expandable navbar (sidebar) example in Mantine UI?

Probably just a button to smoothly minimise the navbar into its icons, and icons should be taken from NavLinks.

Expanded:
![Screenshot 2023-08-02 at 06 03 28](https://github.com/mantinedev/mantine/assets/22228431/b1a1b757-0a59-442f-ad59-37914f6c9fcf)

Minimised:
![Screenshot 2023-08-02 at 06 04 51](https://github.com/mantinedev/mantine/assets/22228431/199d9b13-97e8-4408-b6d0-f3a7e3c80d51)

**Comment by JasonWeise** [comment]
Yes, this would be great! I am trying to implement this but having a terrible time.

**Reply by Grantsuu** [reply]
Same here would love to see this feature added or at least see an example of how to implement it for myself.

**Reply by JudetheGemini** [reply]
This would be great. Having a tough go trying to implement.

**Comment by dongliangCode** [comment]
Have you found the example？

**Reply by JudetheGemini** [reply]
Closest I've seen is this [Here](https://ui.mantine.dev/component/double-navbar/)

**Comment by atumas-bananamilk** [comment]
@rtivital maybe you have some update? most libraries I've checked have it.

**Comment by vovasalyha** [comment]
@atumas-bananamilk The most obvious solution IMO is to utilize `NavLink`'s  `leftSection` and `label` props and simply remove `label` text on toggle. Also change navbar's width on toggle. To make it smooth add some CSS transition on width change and that's it. The only pitfall is that in order to support mobile type of sidebar (drawer) you need more logic, but for PC/laptop screens it should work just fine. Here is a basic example of what I mean: https://codesandbox.io/p/sandbox/mantine-collapsible-sidebar-42jt2m

**Reply by atumas-bananamilk** [reply]
This is good.
Also, what about subsections? How do they expand on a minimised sidebar?
![Screenshot 2024-10-11 at 12 23 17](https://github.com/user-attachments/assets/e9ff689c-3c56-4ed8-b42c-0b97e2ae1d73)

**Reply by rasyidf** [reply]
in my implementation, when collapsed, expanding the minimized sidebar will pop a menu

**Comment by markussen145** [comment]
I support this one. It would make the AppShell more compatible with mobile devices.

**Comment by SqlOwl** [comment]
Please implement this feature I would really love it for my project

**Comment by KGBuApparata** [comment]
WE NEED THIS!!!

**Comment by leomerida15** [comment]
Attached is a very simple collapsible sidebar component that I mounted with mantine, as a suggestion



`typescript
"use client";
import logo from "@/public/logo/logo.svg";
import { Button, Group, Image } from "@mantine/core";
import {
	IconBooks,
	IconChevronRight,
	IconHome2,
	IconLogout,
	IconSwitchHorizontal,
} from "@tabler/icons-react";
import NextImage from "next/image";
import { useMemo, useState } from "react";
import classes from "./NavbarSimple.module.css";

const data = [
	{ icon: IconHome2, label: "Home", link: "" },
	{ icon: IconBooks, label: "catalogo", link: "" },
];

export function Sidebar() {
	const [active, setActive] = useState("Billing");
	const [expanded, setExpanded] = useState(false);

	const links = data.map((item) => {
		const Label = useMemo(() => expanded && <span>{item.label}</span>, [item.label, expanded]);

		return (
			<a
				className={classes.link}
				data-active={item.label === active || undefined}
				href={item.link}
				key={item.label}
				onClick={(event) => {
					event.preventDefault();
					setActive(item.label);
				}}
			>
				<item.icon className={classes.linkIcon} stroke={1.5} />

				{Label}
			</a>
		);
	});

	return (
		<nav className={`${classes.navbar} ${expanded ? classes.expanded : ""} h-full w-auto`}>
			<div
				className={`${classes.navbarMain} relative h-full ${
					expanded ? "w-60" : "w-20"
				} transition-width duration-500`}
			>
				<Group className={classes.header} justify="space-between">
					<Image
						component={NextImage}
						loading="lazy"
						radius="md"
						h={49}
						w="auto"
						fit="contain"
						alt={"logo"}
						src={logo}
					/>
					<Button
						className="absolute right-[-52px] top-[73%] transform"
						onClick={() => setExpanded((expanded) => !expanded)}
						variant="gradient"
					>
						<IconChevronRight
							className={`${
								expanded ? "rotate-180" : ""
							} transition-transform duration-300`}
						/>{" "}
					</Button>
				</Group>
				{links}
			</div>

			<div className={classes.footer}>
				<a href="#" className={classes.link} onClick={(event) => event.preventDefault()}>
					<IconSwitchHorizontal className={classes.linkIcon} stroke={1.5} />
					{expanded && <span>Change account</span>}
				</a>

				<a href="#" className={classes.link} onClick={(event) => event.preventDefault()}>
					<IconLogout className={classes.linkIcon} stroke={1.5} />
					{expanded && <span>Logout</span>}
				</a>
			</div>
		</nav>
	);
}
`
.navbar {
	padding: var(--mantine-spacing-md);
	display: flex;
	flex-direction: column;
	border-right: 1px solid light-dark(var(--mantine-color-gray-3), var(--mantine-color-dark-4));
}

.navbarMain {
	flex: 1;
}

.header {
	padding-bottom: var(--mantine-spacing-md);
	margin-bottom: calc(var(--mantine-spacing-md) * 1.5);
	border-bottom: 1px solid light-dark(var(--mantine-color-gray-3), var(--mantine-color-dark-4));
}

.footer {
	padding-top: var(--mantine-spacing-md);
	margin-top: var(--mantine-spacing-md);
	border-top: 1px solid light-dark(var(--mantine-color-gray-3), var(--mantine-color-dark-4));
}

.link {
	display: flex;
	align-items: center;
	text-decoration: none;
	font-size: var(--mantine-font-size-sm);
	color: light-dark(var(--mantine-color-gray-7), var(--mantine-color-dark-1));
	padding: var(--mantine-spacing-xs) var(--mantine-spacing-sm);
	border-radius: var(--mantine-radius-sm);
	font-weight: 500;

	@mixin hover {
		background-color: light-dark(var(--mantine-color-gray-0), var(--mantine-color-dark-6));
		color: light-dark(var(--mantine-color-black), var(--mantine-color-white));

		.linkIcon {
			color: light-dark(var(--mantine-color-black), var(--mantine-color-white));
		}
	}

	&[data-active] {

		&,
		&:hover {
			background-color: var(--mantine-color-blue-light);
			color: var(--mantine-color-blue-light-color);

			.linkIcon {
				color: var(--mantine-color-blue-light-color);
			}
		}
	}
}

.linkIcon {
	color: light-dark(var(--mantine-color-gray-6), var(--mantine-color-dark-2));
	margin-right: var(--mantine-spacing-sm);
	width: 25px;
	height: 25px;
}

/* NavbarSimple.module.css */
.navbar {
	position: relative;
	transition: width 0.3s;
}



.expandButton {
	position: absolute;
	top: 50%;
	right: -20px;
	/* Ajusta este valor según tus necesidades */
}

.iconRotated {
	transition: width 0.3s;
	transform: rotate(180deg);
}

.header {
	position: relative;
}

.expandButton {
	position: absolute;
	right: -52px;
	top: 73%;
}
`

`

**Reply by SqlOwl** [reply]
Thank you I will give it a try

**Reply by leomerida15** [reply]
@SqlOwl did it work for you?

**Reply by SqlOwl** [reply]
Partially. We are building new shell right now. I hope when it is finished we will share it

**Reply by atumas-bananamilk** [reply]
Doesn't look "very simple"..

**Reply by lwrdr** [reply]
> Partially. We are building new shell right now. I hope when it is finished we will share it

I've been looking for the same thing. Did you have any luck with it?

**Comment by nepton-mattpike** [comment]
Yeah just some basic expand+fade animations on any tree+list style components would be wonderful and inline with the basic animations that appear in other components, it just feels so wrong for a modern UI not to have animations on these things. Thanks!

**Comment by MarwanEnnajih** [comment]
Late to the party, but here's what I wanted to achieve the same result in collapsing the navbar, but not completely because then I want to only show icons. I've found the styles and their properties affecting the navbar when collapse and here's the solution I found. It's important to change the main and navbar style properties to achieve good results. It's not perfect and the animation is a big wonky but I'm sure someone can tweak it a bit to be better.

My navbar width was set to 300.

![Screenshot 2025-04-10 at 12 17 21 PM](https://github.com/user-attachments/assets/8943ee62-39c0-44fb-849f-8f6861c9db20)

All that would be left is to adjust your buttons to show only icons

**Reply by MarwanEnnajih** [reply]
This actually seems to be a better solution since the width of the navbar gets changed. In the picture above I simply translate and set it off screen.

![Screenshot 2025-04-10 at 12 29 54 PM](https://github.com/user-attachments/assets/1e77da87-c566-4562-9bfe-59f0f19fc549)

**Comment by ayovev** [comment]
https://github.com/orgs/mantinedev/discussions/7143
